### PR TITLE
Fix RANSAC for invalid model estimation and confidence corner case

### PR DESCRIPTION
**Pull request by ahojnnes**
See commit description

### CODE REVIEW

**Comment by ahojnnes** [comment]
This should go into a possible 0.11.1 release.

**Comment by JDWarner** [comment]
Held up by a doctest not expecting `True` to be returned.

**Comment by ahojnnes** [comment]
This should fix it. The changes should be fully backwards compatible.

File: skimage/measure/fit.py [hunk 1]
```diff
@@ -482,21 +499,31 @@ def _dynamic_max_trials(n_inliers, n_samples, min_samples, probability):
...
         Probability (confidence) that one outlier-free sample is generated.
-
     Returns
     -------
     trials : int
         Number of trials.
-
     """
+    if n_inliers == 0:
+        return float('inf')
```
Comment (stefanv) [inline skimage/measure/fit.py]: Perhaps use `np.inf` or `np.infty` in the future?  Not important for now.

**Pull request merged**